### PR TITLE
Move triggers into their own handlers

### PR DESCRIPTION
**Pull request by camertron**
Applying the handler pattern to the push and pull triggers. Buys us better error reporting.

@lumoslabs/platform

### CODE REVIEW

File: lib/txgh/handlers/triggers/handler.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        end
+
+        def resource
+          @resource ||= tx_config.resource(resource_slug)
+        end
+
+        def tx_config
+          @tx_config ||= Txgh::Config::TxManager.tx_config(project, repo, branch)
+        end
+
```
Comment (zvkemp) [inline lib/txgh/handlers/triggers/handler.rb:62]: [nitpick.level = 10] empty line!

File: lib/txgh/handlers/triggers/handler.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+              project: config.transifex_project,
+              repo: config.github_repo,
+              branch: request.params.fetch('branch'),
+              resource_slug: request.params.fetch('resource_slug'),
+              logger: logger
+            )
+          end
+        end
+
+        include ResponseHelpers
```
Comment (zvkemp) [inline lib/txgh/handlers/triggers/handler.rb]: could define this on `ResponseHelpers` to only include it once and make it available as class or instance methods:

``` ruby
def self.included(base)
  base.extend(self)
end
```

(if you want)

**Comment by zvkemp** [comment]
Few very minor comments; address them if you feel like it. I like how this is laid out. +1!

File: lib/txgh/handlers/triggers/handler.rb (hunk 2 above)
Comment (camertron) [inline lib/txgh/handlers/triggers/handler.rb]: OooOoOOOOoo I like that!

**Pull request merged**